pre_commit/clientlib.py

```diff
@@ -112,6 +112,24 @@ def validate_manifest_main(argv: Optional[Sequence[str]] = None) -> int:
 META = 'meta'
 
 
+class CheckMutableRev:
```

Comment:
can you use the pattern I adjusted your code to in the other PR?  I believe this should be a `Conditional` similar to the one in `MigrateShaToRev` (which will eventually be deleted)

Reply:
i didn't change it to inherit from `cfgv.Conditional` but took a similar approach as `MigrateShaToRev`, as an inheritance would require passing in the arguments `key`, `check_fn`, `condition_key` and `condition_value` which should be hard coded in `_cond` in my opinion